### PR TITLE
LCFS - Update Fuel Code PyTests and Align Test Filenames #938, #1137

### PR DESCRIPTION
**Pull request by prv-proton**
- #938
- Fixed fuel code bugs - #1137
- Reverting cloning of fuel code that was missed during ag-grid refactor.
- remove hard coding of id fields for status and other fields

### CODE REVIEW

File: backend/lcfs/web/api/fuel_code/repo.py [hunk 1]
```diff
@@ -53,30 +47,17 @@ async def get_fuel_types(self) -> List[FuelType]:
     @repo_handler
     async def get_formatted_fuel_types(self) -> List[Dict[str, Any]]:
         """Get all fuel type options with their associated fuel categories and fuel codes"""
-        # Define the filtering conditions for fuel codes
```
Comment (dhaselhan) [inline backend/lcfs/web/api/fuel_code/repo.py:56]: I read the comments but didn't see a mention why this is being removed. Is this not working correctly or is there validation for expired fuel codes happening elsewhere?

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: backend/lcfs/web/api/fuel_code/views.py [hunk 2]
```diff
@@ -162,53 +141,9 @@ async def delete_fuel_code(
...
-    service: FuelCodeServices = Depends(),
-):
-    """Endpoint to get UCI's"""
-    return await service.get_use_of_a_carbon_intensities()
-
-
 @router.post(
     "/save",
-    response_model=Union[FuelCodeSchema, DeleteFuelCodeResponseSchema],
-    status_code=status.HTTP_200_OK,
```
Comment (dhaselhan) [inline backend/lcfs/web/api/fuel_code/views.py:211]: Whats the reason for removing the status_code? What do these actually do?

File: backend/pyproject.toml [hunk 3]
```diff
@@ -42,9 +42,11 @@ bump-pydantic = "^0.8.0"
 black = "^24.4.2"
 boto3 = "^1.35.26"
 typing-extensions = "^4.12.2"
+pytest-asyncio = "^0.24.0"
 
 [tool.poetry.dev-dependencies]
-pytest = "^7.2.1"
+faker = "^21.0.0"
```
Comment (dhaselhan) [inline backend/pyproject.toml:48]: We recently deleted faker as it was breaking the deployed project, why are we adding it back? I don't see it used in your MR.

File: frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx [hunk 4]
```diff
@@ -208,6 +240,37 @@ const AddFuelCodeBase = () => {
     },
     [saveRow]
   )
+  useEffect(() => {
```
Comment (dhaselhan) [inline frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx:243]: This is a lot of new functionality going into a ticket about writing tests, is this a bug fix?

**Review by dhaselhan — CHANGES_REQUESTED** [review]
I see a lot of changes in this code that are not test related, just want to understand what is going on.

File: backend/pyproject.toml (hunk 3 above)
Comment (prv-proton) [reply backend/pyproject.toml:48]: removed

File: frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx (hunk 4 above)
Comment (prv-proton) [reply frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx:243]: Yes, this is a bug fix. This functionality was initially requested via https://github.com/orgs/bcgov/projects/137/views/1?filterQuery=fuel+code&pane=issue&itemId=55073882&issue=bcgov%7Clcfs%7C433 but during ag-grid editor refactoring this was missed out. Hence adding this functionality back.

File: backend/lcfs/web/api/fuel_code/views.py (hunk 2 above)
Comment (prv-proton) [reply backend/lcfs/web/api/fuel_code/views.py:211]: fixed.

File: frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx (hunk 4 above)
Comment (prv-proton) [reply frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx:243]: raised the bug for this after confirming from Lindsy.
https://github.com/bcgov/lcfs/issues/1137

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**